include/bluetooth/adv_prov/fast_pair.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef BT_ADV_PROV_FAST_PAIR_H_
+#define BT_ADV_PROV_FAST_PAIR_H_
+
+#include <bluetooth/services/fast_pair.h>
+
+/**
+ * @defgroup bt_adv_prov_fast_pair Fast Pair advertising data provider API
+ * @brief Fast Pair advertising data provider API
+ *
+ * @{
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** Set Fast Pair advertising mode.
+ *
+ * User shall make sure that advertising mode is not modified while Fast Pair advertising data
+ * provider is providing advertising data.
+ *
+ * @param[in] fp_adv_mode	Fast Pair advertising mode.
+ */
+void bt_adv_prov_fast_pair_mode_set(enum bt_fast_pair_adv_mode fp_adv_mode);
```

Comment:
According to the argument that you presented in the initial PR with this module, we should not add advanced (configurable) providers as general NCS Bluetooth modules:

> The providers available in the NCS implement a default behavior which could be useful in some simple use-cases. If user needs some more advanced control, he can provide his own implementation instead of enabling provider from the NCS. I prefer this approach instead of making NCS providers configurable (such provider would always have some limitations).

https://github.com/nrfconnect/sdk-nrf/pull/8328#discussion_r944325948

Consider making it a sample provider like the UUID16 provider.

And if we want to keep it as a general NCS Bluetooth module, wouldn't it be better to integrate the advertising provider code with the Fast Pair modules? Of course, it should be possible to turn off such a provider with Kconfig

Reply:
I would still say, that the implementation is a default one (it supports 2 use-cases: using one bond per Bluetooth local identity and allowing application to control Fast Pair pairing mode manually). The exposed API greatly increases number of covered use-cases and gives application more freedom to control behavior. On the other hand, automatic switching between advertising modes is useful e.g. for CAF BLE adv use-case (there are many use-cases where BLE device is supposed to have one owner).